### PR TITLE
Avoid copy in background model evaluation

### PR DESCRIPTION
**Pull request by QRemy**
Avoid copy in background model evaluation in order to slightly improve performance.

### CODE REVIEW

File: gammapy/datasets/map.py [hunk 1]
```diff
@@ -408,12 +409,17 @@ def npred_background(self):
             Predicted counts from the background.
         """
         background = self.background
-
         if self.background_model and background:
             values = self.background_model.evaluate_geom(geom=self.background.geom)
-            background = background * values
+            if self._background_cached is None:
+                self._background_cached = background * values
```
Comment (adonath) [inline gammapy/datasets/map.py:415]: This caching does not make sense to me. Don't you want to check on the model parameters and avoid the recomputation if not needed?

File: gammapy/modeling/models/cube.py [hunk 2]
```diff
@@ -767,8 +768,9 @@ def evaluate(self):
             Background evaluated on the Map
         """
         value = self.spectral_model(self.energy_center).value
-        back_values = self.map.data * value
-        return self.map.copy(data=back_values)
+        self._evaluation_cache.data = self.map.data * value
```
Comment (adonath) [inline gammapy/modeling/models/cube.py:771]: I'd say we don't need the caching on the `.evaluate()` level, this is only used for independent evaluation and not used during fitting.

Comment (QRemy) [reply gammapy/modeling/models/cube.py:771]: for the `TemplateNPredModel` the evaluator call `.evaluate()` 
https://github.com/gammapy/gammapy/blob/19da7840b81b88092d42cd5ea032a7f9a9b7276f/gammapy/datasets/evaluator.py#L361

Comment (adonath) [reply gammapy/modeling/models/cube.py:771]: Yes, but we have caching on the `MapEvaluator`, so the effect on the performance should be completely negligible.

File: gammapy/datasets/map.py (hunk 1 above)
Comment (QRemy) [reply gammapy/datasets/map.py:415]: yes that could be added for now I just did that as most of the time is spend in copy during `background = background * values` just avoiding this allow to reduce npred_background computation by about 40%.

File: gammapy/modeling/models/cube.py (hunk 2 above)
Comment (QRemy) [reply gammapy/modeling/models/cube.py:771]: True

Comment (QRemy) [reply gammapy/modeling/models/cube.py]: removed

File: gammapy/datasets/map.py (hunk 1 above)
Comment (QRemy) [reply gammapy/datasets/map.py]: added

**Comment by adonath** [comment]
Thanks @QRemy, do you have a rough estimate how much this improve the performance in our benchmarks?

**Comment by QRemy** [comment]
> Thanks @QRemy, do you have a rough estimate how much this improve the performance in our benchmarks?

I tested it only on the `analysis_3d_joint` benchmark, it reduces `npred_background` computation time by about 50% (34s to 17s on my machine)

**Review by adonath — APPROVED** [review]
Thanks @QRemy, no further comments from my side.

**Pull request merged**